lib/workers/repository/update/pr/changelog/release-notes.ts

```diff
@@ -419,36 +419,32 @@ export async function addReleaseNotes(
     return input ?? null;
   }
   const output: ChangeLogResult = { ...input, versions: [] };
-  const { repository, sourceDirectory } = input.project;
-  const cacheNamespace = `changelog-${input.project.type}-notes@v2`;
-  function getCacheKey(version: string): string {
-    return `${repository}:${
-      sourceDirectory ? `${sourceDirectory}:` : ''
-    }${version}`;
-  }
+
+  const { repository, sourceDirectory, type: projectType } = input.project;
+  const cacheNamespace = `changelog-${projectType}-notes@v2`;
+  const cacheKeyPrefix = sourceDirectory
+    ? `${repository}:${sourceDirectory}`
+    : `${repository}`;
+
   for (const v of input.versions) {
     let releaseNotes: ChangeLogNotes | null | undefined;
-    const cacheKey = getCacheKey(v.version);
+    const cacheKey = `${cacheKeyPrefix}:${v.version}`;
     releaseNotes = await packageCache.get(cacheNamespace, cacheKey);
-    // istanbul ignore else: no cache tests
-    if (!releaseNotes) {
-      releaseNotes = await getReleaseNotesMd(input.project, v);
-      // istanbul ignore else: should be tested
-      if (!releaseNotes) {
-        releaseNotes = await getReleaseNotes(input.project, v, config);
-      }
-      // Small hack to force display of release notes when there is a compare url
-      if (!releaseNotes && v.compare.url) {
-        releaseNotes = { url: v.compare.url, notesSourceUrl: '' };
-      }
-      const cacheMinutes = releaseNotesCacheMinutes(v.date);
-      await packageCache.set(
-        cacheNamespace,
-        cacheKey,
-        releaseNotes,
-        cacheMinutes
-      );
+    releaseNotes ??= await getReleaseNotesMd(input.project, v);
+    releaseNotes ??= await getReleaseNotes(input.project, v, config);
+
+    // Small hack to force display of release notes when there is a compare url
```

Comment:
@rarkins Do we still need this? If so, what this hack exactly does?

Reply:
No idea, please check the git history for hints

Reply:
Checked, it was 2017, still have no idea. Will try to run with and without it.

Reply:
Anyway, I think we could merge this PR without taking in account this aspect